---
title: "TLR5_Genetics"
author: "SteveYounkin"
date: "November 14, 2015"
output: html_document
---

##Analysis Plan

We will analyze TLR5 genotypes using a robust, R-based reproducible research approach. 

We will keep all of our work in one R Markdown document: TLR5_Genetics.Rmd.  

To communicate with one another as we build this document, we will use the SORCS2 repository in my github account (https://github.com/StevenGYounkin/TLR5), and we will all use git (GitHub gui) for version control. 

All files used or generated by TLR5_Genetics.Rmd will be in this TLR5 repo. 

Joseph will take responsiblity for providing whatever help may be needed for each of us to interact effectively with one another on GitHub to produce a final document.

Our goal, which we may or may not achieve, will be to have an analysis in the Rmd file completed by 12/4/15 that is ready to accept additional data from the ADSP dataset for final analysis and publication.

###What each participant needs:
1. RStudio installed 
2. GitHub installed
3. Local clone of the repo at https://github.com/StevenGYounkin/TLR5
    + TLR5_Genetics.Rmd, the file we are generating together, is in this repo
        + All files used or generated by TLR5_Genetics.Rmd are in this repo
4. R packages installed
    + dplyr
    + data.table
5. PLINK installed and set up so it can be called by R with system("plink......")
6. Perl installed and set up so it can be called by R with system("   ")

###Useful documentation:
1. GitHub
    + https://training.github.com/kit/downloads/github-git-cheat-sheet
2. R Markdown
    + https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf
3. dplyr and tidyr
    + https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf
4. PLINK
    + http://pngu.mgh.harvard.edu/~purcell/plink/
    
###Responsibility for Specific Analysis Steps:
1. Steve will read in existing well-documented TLR5 dataset(s) and show some exploratory data on them
2. Steve will provide the initial Rmd file which will have useful "template" information
3. Ben and Leonie will develop the Rmd file to meet the goals of our Analysis Plan.
    + Focus intially on TLR5 missense SNPs with MAF >=1%
         + "rs56243703","rs5744168","rs5744174","rs5744175","rs5744177"
    + Create Figure 1 showing a TLR5 gene diagram with information on SNP location etc.
    + Revise template Rmd as needed to run analysis on 5 missense SNPs with MAF >= 1%.

###Steve's Content

```{r, echo=T, eval=T}

        #load requisite packages       
library(data.table)
suppressWarnings(suppressMessages(library(dplyr)))

list.files()

        #read in and explore Joseph's ADSP TLR5 flat file        
data <- read.table("TLR5_flat_file.txt", header=T, stringsAsFactors=F)
str(data); head(data);tail(data)
snps <- c("rs56243703","rs5744168","rs5744174","rs5744175","rs5744177")
call <- c(11,12,22)
x <- data %>% filter (SNP_ID %in% snps, Call %in% call) %>% group_by(SNP_ID,Call) %>% summarize (n())
print.data.frame(x) 

        #read in and explore Ydb TLR5 flat file        
data <- read.table("TLR5.csv", sep=",",header=T, stringsAsFactors=F)
str(data); head(data);tail(data)
snps <- c("rs56243703","rs5744168","rs5744174","rs5744175","rs5744177")
call <- c(11,12,22)
x <- data %>% filter (rs %in% snps, Call %in% call) %>% group_by(rs,Call) %>% summarize (n())
print.data.frame(x) 
```


###Ben and Leonie's Content
The data we have on the 5 common TLR5 missense variants in the Younkin database (YdB) were captured with a query that Steve will provide.
The results of this query were copied into TLR5.csv. 

####Create input files from Ydb with (i) all 5 common missense TLR5 SNPs and (ii) the 4 common missense SNPs in ADSP dataset

```{r, echo=T, eval=T}

        #load requisite packages       
        library(data.table)
        suppressWarnings(suppressMessages(library(dplyr)))
        
        #Read in all TLR5 data from Ydb
        data <- read.table("TLR5.csv", sep=',', stringsAsFactors = FALSE, header=T)
        
        #Select only 5 common missense snps, check, write new.csv file
        snps5 <- c("rs56243703","rs5744168","rs5744174","rs5744175","rs5744177")
        data5 <- data %>% filter(rs %in% snps5)
        ck5 <- data5 %>% group_by(rs) %>% summarize(n=n())
        print.data.frame(ck5)
        write.table(data5,"TLR5.5SNPs.Ydb.csv",sep=',',quote=FALSE,row.names=F,col.names=T)
        ck5t <- read.table("TLR5.5SNPs.Ydb.csv",sep=',', stringsAsFactors = FALSE, header=T)
        str(ck5t)
        
        #Select only 4 common missense snps, check, write new.csv file
        snps4 <- c("rs56243703","rs5744174","rs5744175","rs5744177")
        data4 <- data %>% filter(rs %in% snps4)
        ck4 <- data4 %>% group_by(rs) %>% summarize(n=n())
        print.data.frame(ck4)
        write.table(data4,"TLR5.4SNPs.Ydb.csv",sep=',',quote=FALSE,row.names=F,col.names=T)
        ck4t <- read.table("TLR5.4SNPs.Ydb.csv",sep=',',stringsAsFactors = FALSE, header=T)
        str(ck4t)
        
```

####Function (ydb2plink) to read in and analyze Ydb TLR5 datasetTo improve readability of the html file, the r code which follows in the .Rmd file should not be echoed in the html file

```{r, eval=T, echo=F}

        #Create function to read in SQL-derived dataset from Younkin database, 
        #make PLINK files (map, fam, cov.txt, and lgen) and analyze them.

ydb2plink <- function(input="TLR5.csv", mhf=0.001, nspns=5, output="TLR5.Ydb") {

        #load requisite packages       
        library(data.table)
        suppressWarnings(suppressMessages(library(dplyr)))

        #To debug, it was convenient to provide the input called by the function here
        #input="TLR5.5SNPs.Ydb.csv"   
        
        #Read in dataset generated from the Younkin database
        data <- read.table(input, sep=',', stringsAsFactors = FALSE, header=T)
        #str(data)

        #Create output files
        
        #To debug, it was convenient to provide the output called by the function here
        #output <- "TLR5.5SNPs.Ydb"
        
        #Generate names for the PLINK files and the .csv files that will be generated
        out1 <- paste(output,sep="",".map")
        out2 <- paste(output,sep="",".fam")
        out3 <- paste(output,sep="","_cov.txt")
        out4 <- paste(output,sep="",".lgen")
        out5 <- paste(output,sep="",".plink.csv")
        out6 <- paste(output,sep="",".haplo.csv")
        out7 <- paste(output,sep="",".homni.csv")
        #print(out1);print(out2);print(out3);print(out4);print(out5)

        #Generate PLINK map file
        map <- select(data,chr, VariantID, VariantName, rs, position)
        #str(map)
        map <- distinct(map) #Remove duplicates
        #str(map)
        #print.data.frame(map)
        map <- filter(map,!rs=="NULL") #Remove Indels (rs=NULL), genotyping was unreliable.
        #print.data.frame(map)
        #snps1 <- c("rs56243703","rs5744168","rs5744174","rs5744175","rs5744177")
        #snps <- c("rs56243703","rs5744174","rs5744175","rs5744177")
        #map <- map %>% filter(rs %in% snps)
        map <- mutate(map,morgans=0)
        map <- rename(map, bp=position)
        map <- select(map,chr,rs,morgans,bp)
        map <- distinct(map)
        #print(map)
        write.table(map,out1,sep='\t',quote=FALSE,row.names=F,col.names=F)

        #Filter, using dplyr package, to include only subjects with: 
        #Dx = AD or CON, Gend = M or F, ApoE = 22,23,24,33,34,or 44 
        #Series = JS,SibPair_White,RS,AUT,Indiana,NW, DxAge >= 65, BestCall = 1
        #Age (at diagnosis for AD/at entry for CON) of 65 or over
        #rs = "rs56243703","rs5744168","rs5744174","rs5744175","rs5744177"
        dx <- c("AD","CON")
        sex <- c("M","F")
        apoe <- c(22,23,24,33,34,44)
        series <- c("JS", "RS", "AUT", "Indiana", "SibPair_White", "NW")
        data <- filter(data, Dx %in% dx, Gend %in% sex, ApoE %in% apoe, Series %in% series, BestCall == 1, rs %in% snps)
        data <-suppressWarnings(mutate(data,Age=as.numeric(DxAge)))
        data <- filter (data, Age >= 65)
        x <- data %>% group_by(BestCall,Series,Gend,Dx) %>% summarize(n=n())
        #print(x)
        y <- data %>% group_by(ApoE) %>% summarize(n=n())
        #print(y)
        z <- data %>% select(SubKey,Dx,Gend,DxAge,ApoE,BestCall) %>% arrange(as.numeric(DxAge))
        #str(z)
        z <- distinct(z)
        #str(z)
        #head(z)
        #tail(z)
                
        #Create columns for PLINK fam, cov, and lgen files
        data <- mutate(data, AFF=99, A1=99, A2=99, APOE4dose=99, APOE2dose=99, SEX=99, JS=0, RS=0, AUT=0, NCRAD=0,NW=0, FID=0, IID=SubKey, PID=0, MID=0)
        
        #Populate variables, making sure to hew to PLINK convention

        data[data$Dx == "AD",][, "AFF"] <- 2
        data[data$Dx == "CON",][, "AFF"] <- 1

        data[data$Call == 11,][, "A1"] <- 1
        data[data$Call == 12,][, "A1"] <- 1
        data[data$Call == 22,][, "A1"] <- 2

        data[data$Call == 11,][, "A2"] <- 1
        data[data$Call == 12,][, "A2"] <- 2
        data[data$Call == 22,][, "A2"] <- 2

        data[data$Gend == "M",][, "SEX"] <- 1
        data[data$Gend == "F",][, "SEX"] <- 2

        data[data$ApoE == 22,][, "APOE4dose"] <- 0
        data[data$ApoE == 23,][, "APOE4dose"] <- 0
        data[data$ApoE == 24,][, "APOE4dose"] <- 1
        data[data$ApoE == 33,][, "APOE4dose"] <- 0
        data[data$ApoE == 34,][, "APOE4dose"] <- 1
        data[data$ApoE == 44,][, "APOE4dose"] <- 2

        data[data$ApoE == 22,][, "APOE2dose"] <- 2
        data[data$ApoE == 23,][, "APOE2dose"] <- 1
        data[data$ApoE == 24,][, "APOE2dose"] <- 1
        data[data$ApoE == 33,][, "APOE2dose"] <- 0
        data[data$ApoE == 34,][, "APOE2dose"] <- 0
        data[data$ApoE == 44,][, "APOE2dose"] <- 0

        data[data$Series == "JS",][, "JS"] <- 1
        #data[data$Series == "SibPair_White",][, "JS"] <- 1
        data[data$Series == "RS",][, "RS"] <- 1
        data[data$Series == "AUT",][, "AUT"] <- 1
        #data[data$Series == "Indiana",][, "NCRAD"] <- 1
        #data[data$Series == "NW",][, "NW"] <- 1

        #Check coding for PLINK during debugging, using dplyr as exemplified below
        #print(data %>% group_by(rs,AFF,SEX,APOE4dose, APOE2dose,A1,A2) %>% summarize(n=n()))
        #print.data.frame(data %>% group_by(rs,A1,A2) %>% summarize(n=n()))
        #print.data.frame(data %>% group_by(rs,AFF) %>% summarize(n=n()))
        #print.data.frame(data %>% group_by(rs,SEX) %>% summarize(n=n()))
        #print.data.frame(data %>% group_by(APOE4dose,APOE2dose) %>% summarize(n=n()))      
        #print.data.frame(y)

        #Create PLINK fam file, commands commented out were useful during debugging
        fam <- select(data,FID,IID,PID,MID,SEX,AFF)
        fam <- distinct(fam)
        #str(fam); head(fam); tail(fam)
        write.table(fam,out2,sep='\t',quote=FALSE,row.names=F,col.names=F)
        #SORCS2fam=read.table("SORCS2.fam", header = F)
        #str(SORCS2fam);head(SORCS2fam);tail(SORCS2fam)

        #Create PLINK cov file, commands commented out were useful during debugging
        #Not sure why, but errors sometimes occurred in r when using "SORCS2.cov"" as output using write.table
        #So I used "SORCS2cov.txt" thereby avoiding this problem
        cov <- select(data,FID,IID,JS,RS,AUT,NCRAD,NW,SEX,Age,APOE4dose,APOE2dose)
        cov <- distinct(cov)
        #str(cov)
        write.table(cov,out3,sep='\t',quote=FALSE,row.names=F,col.names=T)
        #SORCS2cov=read.table("SORCS2cov.txt", header = T)
        #str(SORCS2cov);head(SORCS2cov); tail(SORCS2cov)

        #Create PLINK lgen file
        lgen <- select(data,FID,IID,rs,A1,A2)
        #str(lgen)
        write.table(lgen,out4,sep='\t',quote=FALSE,row.names=F,col.names=F)
        #SORCS2lgen=read.table("SORCS2.lgen", header = F)
        #str(SORCS2lgen)

        #Analyse the files in PLINK, format, and print the output. The analysis here is the same as that 
        #in the script for the adsp2plink function below, where it is better annotated.        
        
        assoc <- paste("plink",sep=" ","--lfile", output, "--assoc", "--ci 0.95", "--out", output)
        
        
        logistic <- paste("plink",sep=" ","--lfile", output, "--logistic", "--covar", out3, "--covar-name RS,AUT,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --out", output)
        haplog <- paste("plink",sep=" ","--lfile", output, "--hap-window 20 --hap-logistic", "--covar", out3, "--covar-name RS,AUT,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --out", output)

        assocfile <- paste(output,sep="",".assoc")
        logisfile <- paste(output,sep="",".assoc.logistic")
        haplogfile <- paste(output,sep="",".assoc.hap.logistic")

        system(assoc)
        system(logistic)
        system(haplog)

        f1 <- read.table(assocfile, header=T)
        #print(f1)
        f1 <- f1 %>% select(CHR,SNP,BP,F_A, F_U, OR, L95, U95, P) %>% rename(FREQ_AD=F_A, FREQ_CON=F_U, OR_assoc=OR, L95a=L95, U95a=U95, P_assoc=P)
        print(f1,digits=2,row.names=F)
        print("", quote=F)
        #head(f1)

         f1 <- f1 %>% select(CHR,SNP,BP,FREQ_AD,FREQ_CON,OR_assoc,L95a,U95a,P_assoc)
        f2 <- read.table(logisfile, header=T)
        #print(f2)
        f2 <- f2 %>% filter (TEST=="ADD") %>% select (SNP,NMISS,OR,L95,U95,P) %>% rename(OR_log=OR, P_log=P)
        #head(f2)
        f3 <- left_join(f1,f2,by="SNP")
        f3 <- mutate(f3,Group="1_Ydb")
        options(width = 130)
        print(f3,digits=2,row.names=F)
        print("", quote=F)
        write.table(f3,out5,sep=",", quote=FALSE,row.names=F,col.names=T)
        #print("PLINK commands:");print(assoc);print(logistic)

        f4 <- read.table(haplogfile, header=T)
        f4 <- mutate(f4,Group="1_Ydb")
        f4 <- f4 %>% select(-NSNP,-NHAP,-STAT) %>% arrange(desc(HAPLOTYPE))
        print(f4,digits=2, row.names=F)
        print("", quote=F)
        write.table(f4,out6,sep=",", quote=FALSE,row.names=F,col.names=T)

        hapomni <- paste("plink",sep=" ","--lfile", output, "--hap-window 20 --hap-logistic", "--covar", out3, "--covar-name RS,AUT,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --hap-omnibus --out", output)
        system(hapomni)
        f4 <- read.table(haplogfile, header=T)
        f4 <- f4 %>% rename(Pomni=P)
        f4 <- mutate(f4,Group="1_Ydb")
        print(f4,digits=2, row.names=F)
        print("", quote=F)
        write.table(f4,out7,sep=",", quote=FALSE,row.names=F,col.names=T)
}

```

####Function (adsp2plink) to read in and analyze ADSP TLR5 dataset

The data we have on the 5 common TLR5 missense variants in 3000 subjects from our ADSP dataset were captured by Joseph Reddy. Only 4 of the 5 common TLR5 missens SNPs were found in the ADSP dataset. We put his file (TLR5_flat_file 3000.txt) into the TLR5 repo.The following function reads the data in, creates PLINK files, and then analyzes them in PLINK as for the TLR5 dataset from the Younkin database.To improve readability of the html output, the r code should not be echoed.

```{r, eval=T, echo=F}
        #Create function that (i) reads in the ADSP dataset generated by Joseph Reddy, (ii) outputs PLINK map, fam, cov.txt, and lgen files,
        #(iii) analyzes in PLINK using --assoc, as well as --logistic, and --hap-logistic with appropriate covariates, 
        #(iv) prints out well organized results
        
adsp2plink <- function(input="TLR5_flat_file.txt", output="TLR5.adsp") {

        #load requisite packages       
        library(data.table)
        suppressWarnings(suppressMessages(library(dplyr)))  
        
        #While debugging it is helpful to set input and output values that will be called by the function
        #It is also useful to employ the various str and print commands that are commented out below
        #input <- "TLR5_flat_file 3000.txt"
        #output <- "TLR5_3000.adsp"        
        #Read in data on 4 common SORCS2 missense variants in the ADSP database        
        
        data <- read.table(input, stringsAsFactors = FALSE, header=T)
        #str(data); head(data); tail(data)

        #Create output files
        out1 <- paste(output,sep="",".map")
        out2 <- paste(output,sep="",".fam")
        out3 <- paste(output,sep="","_cov.txt")
        out4 <- paste(output,sep="",".lgen")
        out5 <- paste(output,sep="",".plink.csv")
        out6 <- paste(output,sep="",".haplo.csv")
        out7 <- paste(output,sep="",".homni.csv")
        #print(out1);print(out2);print(out3);print(out4);print(out5)

        #Filter, using dplyr, to include only subjects with: 
        #Case.control = 2 or 1 corresponding to AD or CON, Sex = 0 or 1 (M or F), APOE = 22,23,24,33,34,or 44 
        #Series = ADSP, Age >= 65, Call = 11,12, or 22
        library(dplyr)
        dx_1 <- c(2,1,0)
        sex_1 <- c(0,1)
        apoe_1 <- c(22,23,24,33,34,44)
        series_1 <- c("ADSP")
        call_1 <- c(11,12,22)
        data <- filter(data, Case.control %in% dx_1, Sex %in% sex_1, APOE %in% apoe_1, Series %in% series_1, Call %in% call_1)
        data <- filter (data, Age >= 65)
        #str(data);head(data);tail(data)

        #Create PLINK map file
        map <- select(data, SNP_ID, Position)
        map <- mutate(map,morgans=0, chr=1)
        map <- rename(map, bp=Position, rs=SNP_ID)
        map <- select(map,chr,rs,morgans,bp)
        map <- distinct(map)
        snps1 <- c("rs56243703","rs5744168","rs5744174","rs5744175","rs5744177")
        #snps <- c("rs56243703","rs5744174","rs5744175","rs5744177")
        map <- filter(map,rs %in% snps1 )
        #print(map)
        write.table(map,out1,sep='\t',quote=FALSE,row.names=F,col.names=F)

        #Create variables for PLINK fam, cov, and lgen files
        data <- mutate(data, AFF=99, A1=99, A2=99, APOE4dose=99, APOE2dose=99, SEX=99, ADSP=0, FID=0, IID=as.character(Subject_ID), PID=0, MID=0)

        data[data$Case.control == 0,][, "AFF"] <- 1  #CON=0
        data[data$Case.control == 1,][, "AFF"] <- 2  #AD=1
        data[data$Case.control == 2,][, "AFF"] <- 2  #probable AD=2 are included as AD for PLINK analysis

        data[data$Call == 11,][, "A1"] <- 1  #1 = major allele, 2 = minor allele
        data[data$Call == 12,][, "A1"] <- 1  #split the allele calls as required by PLINK
        data[data$Call == 22,][, "A1"] <- 2  

        data[data$Call == 11,][, "A2"] <- 1
        data[data$Call == 12,][, "A2"] <- 2
        data[data$Call == 22,][, "A2"] <- 2

        data[data$Sex == 0,][, "SEX"] <- 1   #M=0
        data[data$Sex == 1,][, "SEX"] <- 2   #F=1

        data[data$APOE == 22,][, "APOE4dose"] <- 0 #code dose of A4 allele, 0,1,or2
        data[data$APOE == 23,][, "APOE4dose"] <- 0
        data[data$APOE == 24,][, "APOE4dose"] <- 1
        data[data$APOE == 33,][, "APOE4dose"] <- 0
        data[data$APOE == 34,][, "APOE4dose"] <- 1
        data[data$APOE == 44,][, "APOE4dose"] <- 2

        data[data$APOE == 22,][, "APOE2dose"] <- 2 #code dose of A4 allele, 0,1,or2
        data[data$APOE == 23,][, "APOE2dose"] <- 1
        data[data$APOE == 24,][, "APOE2dose"] <- 1
        data[data$APOE == 33,][, "APOE2dose"] <- 0
        data[data$APOE == 34,][, "APOE2dose"] <- 0
        data[data$APOE == 44,][, "APOE2dose"] <- 0

        data[data$Series == "ADSP",][, "ADSP"] <- 1 #1 indicates subject is ADSP, otherwise would be 0
        #For debugging, evaluate whether data has been formatted correctly by filtering and grouping appropriately
        #y <- data %>% filter(SNP_ID=="snps") %>% group_by(SNP_ID,AFF,A1,A2,SEX,ADSP,FID) %>% summarize(n=n())
        #print(y)

        #Create PLINK fam file
        fam <- select(data,FID,IID,PID,MID,SEX,AFF)
        fam <- distinct(fam)
        #str(fam); head(fam); tail(fam)
        write.table(fam,out2,sep='\t',quote=FALSE,row.names=F,col.names=F)
        #TLR5fam_1=read.table("TLR5.adsp.fam", header = F)
        #str(TLR5fam_1);head(TLR5fam_1);tail(TLR5fam_1)

        #Create PLINK cov file; 
        #Not sure why, but errors sometimes occurred in r when using "TLR5.cov"" as output using write.table
        #So I used "TLR5cov.txt" thereby avoiding this problem
        cov <- select(data,FID,IID,ADSP,SEX,Age,APOE4dose,APOE2dose)
        cov <- distinct(cov)
        #For debugging make sure structure looks good
        #str(cov)
        write.table(cov,out3,sep='\t',quote=FALSE,row.names=F,col.names=T)
        #For debugging read table back in and make sure it looks right
        #TLR5cov_1=read.table("TLR5_1cov.txt", header = T)
        #str(TLR5cov_1);head(TLR5cov_1); tail(TLR5cov_1)

        #Create PLINK lgen file
        lgen <- select(data,FID,IID,SNP_ID,A1,A2)
        #str(lgen)
        write.table(lgen,out4,sep='\t',quote=FALSE,row.names=F,col.names=F)

        #Create command strings for --assoc, --logistic, and --haplogistic
        assoc <- paste("plink",sep=" ","--lfile", output, "--assoc", "--ci 0.95", "--out", output)
        logistic <- paste("plink",sep=" ","--lfile", output, "--logistic", "--covar", out3, "--covar-name SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--out", output)
        haplog <- paste("plink",sep=" ","--lfile", output, "--hap-window 30 --hap-logistic", "--covar", out3, "--covar-name SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --out", output)

        #Create names for the files that PLINK generates
        assocfile <- paste(output,sep="",".assoc")
        logisfile <- paste(output,sep="",".assoc.logistic")
        haplogfile <- paste(output,sep="",".assoc.hap.logistic")

        #Run the PLINK analyses
        system(assoc)
        system(logistic)
        system(haplog)

        #Read in, format, and print the results from --assoc
        f1 <- read.table(assocfile, header=T)
        #print(f1)
        f1 <- f1 %>% select(CHR,SNP,BP,F_A, F_U, OR, L95, U95, P) %>% rename(FREQ_AD=F_A, FREQ_CON=F_U, OR_assoc=OR, L95a=L95, U95a=U95, P_assoc=P)
        print(f1,digits=2,row.names=F)
        print("", quote=F)
        #head(f1)

        #Read in results for --logistic, combine with results from --logistic, format, print
        f1 <- f1 %>% select(CHR,SNP,BP,FREQ_AD,FREQ_CON,OR_assoc,L95a,U95a,P_assoc)
        f2 <- read.table(logisfile, header=T)
        #print(f2)
        f2 <- f2 %>% filter (TEST=="ADD") %>% select (SNP,NMISS,OR,L95,U95,P) %>% rename(OR_log=OR, P_log=P)
        #head(f2)
        f3 <- left_join(f1,f2,by="SNP")
        f3 <- mutate(f3,Group="2_ADSP")
        options(width = 130)
        print(f3,digits=2,row.names=F)
        print("", quote=F)
        write.table(f3,out5,sep=",", quote=FALSE,row.names=F,col.names=T)

        #Read in results for --hap-logistic, format, print
        f4 <- read.table(haplogfile, header=T)
        f4 <- mutate(f4,Group="2_ADSP")
        f4 <- f4 %>% select(-NSNP,-NHAP,-STAT) %>% arrange(desc(HAPLOTYPE))
        print(f4,digits=2, row.names=F)
        print("", quote=F)
        write.table(f4,out6,sep=",", quote=FALSE,row.names=F,col.names=T)

        #Run and read in results for "--hap-logistic --hap-omnibus"", format, print
        hapomni <- paste("plink",sep=" ","--lfile", output, "--hap-window 30 --hap-logistic", "--covar", out3, "--covar-name SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --hap-omnibus --out", output)
        system(hapomni)
        f4 <- read.table(haplogfile, header=T)
        f4 <- f4 %>% rename(Pomni=P)
        f4 <- mutate(f4,Group="2_ADSP")
        print(f4,digits=2, row.names=F)
        print("", quote=F)
        write.table(f4,out7,sep=",", quote=FALSE,row.names=F,col.names=T)
}
```

####Function (comb2plink) to create combined ydb-adsp dataset and analyse it
```{r, echo=F, eval=T}
comb2plink <- function(output="TLR5.comb") {

                #load requisite packages       
library(data.table)
suppressWarnings(suppressMessages(library(dplyr)))

        #Create output files
output <- "TLR5.comb"
out1 <- paste(output,sep="",".map")
out2 <- paste(output,sep="",".fam")
out3 <- paste(output,sep="","_cov.txt")
out4 <- paste(output,sep="",".lgen")
out5 <- paste(output,sep="",".plink.csv")
out6 <- paste(output,sep="",".haplo.csv")
out7 <- paste(output,sep="",".homni.csv")

        #Generate PLINK map file by copying and renaming "TLR5.adsp.map"
map <- read.table("TLR5.adsp.map", header = F)
print(map)
write.table(map, out1, sep='\t',quote=FALSE,row.names=F,col.names=F)

        #Use dplyr bind_rows to generate combined fam, cov, and lgen files

fam.ydb <- read.table("TLR5.5SNPs.Ydb.fam", header=F, stringsAsFactors = FALSE)
colnames(fam.ydb) <- c("FID","IID","PID","MID","SEX","AFF")
  #Change IID from number to character for compatiblity with fam.adsp
fam.ydb <- fam.ydb %>% mutate(IID=as.character(IID)) 
#str(fam.ydb)
fam.adsp <- read.table("TLR5.adsp.fam", header=F, stringsAsFactors = FALSE)
colnames(fam.adsp) <- c("FID","IID","PID","MID","SEX","AFF")
#str(fam.adsp)
fam.comb <- bind_rows(fam.ydb,fam.adsp)
#str(fam.comb)
write.table(fam.comb, out2, sep='\t',quote=FALSE,row.names=F,col.names=F)

cov.ydb <- read.table("TLR5.5SNPs.Ydb_cov.txt", header=T, stringsAsFactors = FALSE)
cov.ydb <- cov.ydb %>% mutate(IID=as.character(IID))
#str(cov.ydb)
cov.adsp <- read.table("TLR5.adsp_cov.txt", header=T, stringsAsFactors = FALSE)
#str(cov.adsp)
cov.comb <- bind_rows(cov.ydb,cov.adsp)
#str(cov.comb)
x <- cov.comb %>% group_by(JS,RS,AUT,NCRAD,NW,ADSP) %>% summarise(n())
#print(x)
                #Combined series covariates had NAs that should be 0                
cov.comb[is.na(cov.comb)]<-0
x <- cov.comb %>% group_by(JS,RS,AUT,NCRAD,NW,ADSP) %>% summarise(n())
#print(x)
                # NA replaced by 0 in all series covariates
write.table(cov.comb, out3, sep='\t',quote=FALSE,row.names=F,col.names=T)

lgen.ydb <- read.table("TLR5.5SNPs.Ydb.lgen", header=F, stringsAsFactors = FALSE)
colnames(lgen.ydb) <- c("FID","IID","rs","A1","A2")
lgen.ydb <- lgen.ydb %>% mutate(IID=as.character(IID))
#str(lgen.ydb)
lgen.adsp <- read.table("TLR5.adsp.lgen", header=F, stringsAsFactors = FALSE)
colnames(lgen.adsp) <- c("FID","IID","rs","A1","A2")
#str(lgen.adsp)
lgen.comb <- bind_rows(lgen.ydb,lgen.adsp)
#str(lgen.comb)
write.table(lgen.comb, out4, sep='\t',quote=FALSE,row.names=F,col.names=F)

assoc <- paste("plink",sep=" ","--lfile", output, "--assoc", "--ci 0.95", "--out", output)
logistic <- paste("plink",sep=" ","--lfile", output, "--logistic", "--covar", out3, "--covar-name ADSP,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--out", output)
haplog <- paste("plink",sep=" ","--lfile", output, "--hap-window 30 --hap-logistic", "--covar", out3, "--covar-name ADSP,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --out", output)


assocfile <- paste(output,sep="",".assoc")
logisfile <- paste(output,sep="",".assoc.logistic")
haplogfile <- paste(output,sep="",".assoc.hap.logistic")

system(assoc)
system(logistic)
system(haplog)

f1 <- read.table(assocfile, header=T)
#print(f1)
f1 <- f1 %>% select(CHR,SNP,BP,F_A, F_U, OR, L95, U95, P) %>% rename(FREQ_AD=F_A, FREQ_CON=F_U, OR_assoc=OR, L95a=L95, U95a=U95, P_assoc=P)
print(f1,digits=2,row.names=F)
print("", quote=F)
#head(f1)

f1 <- f1 %>% select(CHR,SNP,BP,FREQ_AD,FREQ_CON,OR_assoc,L95a,U95a,P_assoc)
        f2 <- read.table(logisfile, header=T)
        #print(f2)
        f2 <- f2 %>% filter (TEST=="ADD") %>% select (SNP,NMISS,OR,L95,U95,P) %>% rename(OR_log=OR, P_log=P)
        #head(f2)
        f3<- left_join(f1,f2,by="SNP")
        options(width = 130)
#head(f2)
f3 <- left_join(f1,f2,by="SNP")
f3 <- mutate(f3,Group="3_Comb")
print(f3,digits=2,row.names=F)
print("", quote=F)
write.table(f3,out5,sep=",", quote=FALSE,row.names=F,col.names=T)

f4 <- read.table(haplogfile, header=T)
f4 <- mutate(f4,Group="3_Comb")
f4 <- f4 %>% select(-NSNP,-NHAP,-STAT) %>% select(Group,CHR:P) %>% arrange(desc(HAPLOTYPE,Group))
print(f4,digits=2, row.names=F)
print("", quote=F)
write.table(f4,out6,sep=",", quote=FALSE,row.names=F,col.names=T)

hapomni <- paste("plink",sep=" ","--lfile", output, "--hap-window 30 --hap-logistic", "--covar", out3, "--covar-name ADSP,SEX,Age,APOE4dose,APOE2dose", "--ci 0.95", "--mhf 0.007 --hap-omnibus --out", output)
system(hapomni)
f4 <- read.table(haplogfile, header=T)
f4 <- f4 %>% rename(Pomni=P)
f4 <- mutate(f4,Group="3_Comb")
print(f4,digits=2, row.names=F)
write.table(f4,out7,sep=",", quote=FALSE,row.names=F,col.names=T)
#print("", quote=F)

        #Reorganize plink.csv output files to facilitate comparison
f5 <- read.table("TLR5.5SNPs.Ydb.plink.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f5)
f6 <- read.table("TLR5.adsp.plink.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f6)
f7 <- read.table("TLR5.comb.plink.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f7)
f8 <- bind_rows(f5,f6,f7)
f8 <- arrange(f8,BP,Group)
f8 <- select(f8,Group,CHR:OR_assoc,P_assoc:P_log)
print(f8,digits=2, row.names=F)

        #Reorganize haplo.csv output files to facilitate comparison
f5 <- read.table("TLR5.5SNPs.Ydb.haplo.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f5)
f6 <- read.table("TLR5.adsp.haplo.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f6)
f7 <- read.table("TLR5.comb.haplo.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f7)
f8 <- bind_rows(f5,f6,f7)
f8 <- arrange(f8,desc(HAPLOTYPE),Group)
f8 <- select(f8,Group,CHR:P)
print(f8,digits=2, row.names=F)

       #Reorganize haplo.csv output files to facilitate comparison
f5 <- read.table("TLR5.5SNPs.Ydb.homni.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f5)
f6 <- read.table("TLR5.adsp.homni.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f6)
f7 <- read.table("TLR5.comb.homni.csv", sep=',', stringsAsFactors = FALSE, header=T)
#print(f7)
f8 <- bind_rows(f5,f6,f7)
f8 <- arrange(f8,Group)
f8 <- select(f8,Group,NSNP:Pomni)
print(f8,digits=2, row.names=F)
}
```

        #Reorganize TLR5.comb files to facilitate analysis of haplotypes and multilocus genotypes

}
```

####Create plink function
```{r, echo=F, eval=F}

plink <- function(input="TLR5.comb", comstring="plink --lfile  TLR5.comb --assoc --ci 0.95 --out TLR5test", output=TLR5.comb) {
        comstring <- paste("plink",sep=" ","--lfile", input, "--hap-window 30 --hap-phase", "--out", output)
        system(comstring)        
}        
```

                #load requisite packages       
library(data.table)
suppressWarnings(suppressMessages(library(dplyr)))        
ydb2plink(input="TLR5.5SNPs.Ydb.csv", output="TLR5.5SNPs.Ydb")
```

####Analyze 5 common missense SNPS in TLR5 dataset from Ydb
```{r, eval=T}
ydb2plink(input="TLR5.5SNPs.Ydb.csv", output="TLR5.5SNPs.Ydb")
```

####Analyze 4 common missense SNPS in TLR5 dataset from Ydb that are found in ADSP dataset
```{r, eval=F, echo=F}
ydb2plink(input="TLR5.4SNPs.Ydb.csv", output="TLR5.4SNPs.Ydb")
```

####Analyze TLR5 dataset from ADSP
```{r, eval=T}
adsp2plink(input="TLR5_flat_file.txt", output="TLR5.adsp")
```

####Analyze combined (Ydb+ADSP) TLR5 dataset
```{r, eval=T}
comb2plink(output="TLR5.comb")
```


####Compare SORCS2 datasets: Chris Medway vs. current Ydb
```{r, eval=F, echo=F}
        #Compare series in covariate file

cov.ydb<- read.table("SORCS2_Ydb_cov.txt",header=T, stringsAsFactors = FALSE)
colnames(cov.ydb) <- c("FID","IID","JS.ydb","RS.ydb","AUT.ydb","NCRAD.ydb","NW.ydb","SEX.ydb","Age.ydb","APOE4dose.ydb","APOE2dose.ydb")
#str(cov.ydb)
x <- cov.ydb %>% group_by(JS.ydb,RS.ydb,AUT.ydb,NCRAD.ydb,NW.ydb) %>% summarize(n())
#print.data.frame(x,row.names=F)

cov.cm<- read.table("SORCS2.CM_cov.txt",header=T)
#str(cov.cm)
colnames(cov.cm) <- c("FID","IID","JS.cm","RS.cm","AUT.cm","NCRAD.cm","NW.cm","SEX.cm","Age.cm","APOE4dose.cm","APOE2dose.cm")
#str(cov.cm)
x <- cov.cm %>% group_by(JS.cm,RS.cm,AUT.cm,NCRAD.cm,NW.cm) %>% summarize(n())
print.data.frame(x,row.names=F)

       #Full join cm and ydb lgen and cov files

cov.ydb<- read.table("SORCS2_Ydb_cov.txt",header=T, stringsAsFactors = FALSE)
colnames(cov.ydb) <- c("FID","IID","JS.ydb","RS.ydb","AUT.ydb","NCRAD.ydb","NW.ydb","SEX.ydb","Age.ydb","APOE4dose.ydb","APOE2dose.ydb")
#str(cov.ydb)

lgen.head.ydb <- c("FID","IID","rs.ydb","A1.ydb", "A2.ydb")
lgen.ydb<- read.table("SORCS2_Ydb.lgen", header=F, col.names=lgen.head.ydb, stringsAsFactors = FALSE)
#str(lgen.ydb)

fam.head.ydb <- c("FID","IID","PID.ydb","MID.ydb", "SEX.ydb", "DX.ydb")
fam.ydb<- read.table("SORCS2_Ydb.fam", header=F, col.names=fam.head.ydb, stringsAsFactors = FALSE)
#str(fam.ydb)

fam.ydb <- select(fam.ydb,IID,DX.ydb)
all.ydb <- left_join(cov.ydb,fam.ydb,by="IID")
#str(all.ydb)
lgen.ydb <- select(lgen.ydb,IID:A2.ydb)
#str(lgen.ydb)
all.ydb <- left_join(all.ydb, lgen.ydb, by="IID")
#str(all.ydb)

#str(all.ydb)

cov.cm<- read.table("SORCS2.CM_cov.txt",header=T, stringsAsFactors = FALSE)
colnames(cov.cm) <- c("FID","IID","JS.cm","RS.cm","AUT.cm","NCRAD.cm","NW.cm","SEX.cm","Age.cm","APOE4dose.cm","APOE2dose.cm")
str(cov.cm)

lgen.head.cm <- c("FID","IID","rs.cm","A1.cm", "A2.cm")
lgen.cm<- read.table("SORCS2.CM.lgen", header=F, col.names=lgen.head.cm, stringsAsFactors = FALSE)
#str(lgen.cm)

fam.head.cm <- c("FID","IID","PID.cm","MID.cm", "SEX.cm", "DX.cm")
fam.cm<- read.table("SORCS2.CM.fam", header=F, col.names=fam.head.cm, stringsAsFactors = FALSE)
#str(fam.cm)

fam.cm <- select(fam.cm,IID,DX.cm)
all.cm <- left_join(cov.cm,fam.cm,by="IID")
#str(all.cm)
lgen.cm <- select(lgen.cm,IID:A2.cm)
#str(lgen.cm)
all.cm <- left_join(all.cm, lgen.cm, by="IID")
#all.cm <- 
#str(all.cm)


#all.cm.ydb <- full_join(all.cm,all.ydb, by="IID")
#str(all.cm.ydb)
#all.cm.ydb <- filter(all.cm.ydb,rs.cm==rs.ydb | is.na(rs.cm) | is.na(rs.ydb))
#str(all.cm.ydb)
#addl.cm <- filter(all.cm.ydb,is.na(rs.ydb))
#str(addl.cm)
#only.ydb <- filter(all.cm.ydb,!is.na(rs.ydb))
#str(only.ydb)


#addl.cm <- addl.cm %>% select()
#str(all.cm.ydb)
#x <- all.cm.ydb %>% filter(Age.cm==Age.ydb) %>% summarise(n())
#print(x)
#x <- sum(is.na(all.cm.ydb$IID))
#print(x)
#x <- all.cm.ydb %>% select(IID) %>% distinct %>% summarize(n())
#print(x)

#x <- cov.ydb %>% group_by(JS,RS,AUT,NCRAD,NW) %>% summarize(n())
#print(x,row.names=F)
#cov.cm<- read.table("SORCS2.CM_cov.txt",header=T)
#str(cov.cm)
#x <- cov.cm %>% group_by(JS,RS,AUT,NCRAD,NW) %>% summarize(n())
#print(x,row.names=F)
```

####Ben and Leonie's Content for Making Figure 1
```{r, eval=F}
![](20151112 genemap.png)

<span style="color:grey"><font size="0.8">__*Figure 1* | SorCS2 gene diagram. A)__ Position of SorCS2 on chromosome 4 (red box). __B)__ SorCS2 gene intron-exon layout. Exon length 100x magnified, exons containing the SNPs of interest highlighted in green. Zoomed in on relative position of SNPs of interest on exons 7, 16 and 17. __C)__ Relative position of SNPs of interest on SorCS2 protein. Domains highlighted and indicated.</font></span>
```

###Joseph's Content
ADSP data was processed using the Genome GPS pipeline by the Bioinformatics Core in Rochestor. After variant calling (using 'GenotypeGVCFs' in the GATK pipeline), variants were annotated using BioR.  
Variants from genes of interest (here, SORCS2) were identified and processed using Perl scripts to generate files for downstream analysis.

```{r, eval=F}
system("perl generate_flat_file_SG.pl ADSP ADSP_sample_sex_age_AD_APOE.txt SORCS2_variants.txt 1000 SORCS2_ff.txt SORCS2_allele_counts.txt")

#Sex : 0: Male, 1: Female
#Case/Control: 0: Control, 1: AD, 2: Probable AD
#Call: 11/12/22
#Genotype: 0/0, 0/1, 1/1,  [./., ./0, ./1, 0/., 1/. encoded as NA]

#Importing Flat File
SORCS2_FF <- read.table("SORCS2_ff.txt",header = TRUE, check.names = FALSE)
head(SORCS2_FF)

#Importing allele counts for each variant
SORCS2_variants <- read.table("SORCS2_allele_counts.txt", header = TRUE, check.names = FALSE)
head(SORCS2_variants)
```






